example/basic_sample.py

```diff
@@ -17,49 +17,93 @@
+                             "local OneAgent endpoint")
 
-# start_pipeline will notify the MeterProvider to begin collecting/exporting
-# metrics with the given meter, exporter and interval in seconds
-metrics.get_meter_provider().start_pipeline(meter, exporter, 5)
+    parser.add_argument("-t", "--token", default=None, type=str, dest="token",
+                        help="API Token generated in the Dynatrace UI. Needs "
+                             "to have metrics ingestion enabled in order to "
+                             "work correctly. Can be omitted when exporting "
+                             "to the local OneAgent.")
 
-requests_counter = meter.create_counter(
-    name="requests",
-    description="number of requests",
-    unit="1",
-    value_type=int
-)
+    parser.add_argument("-nm", "--no-metadata", dest="metadata_enrichment",
+                        action="store_false",
+                        help="Turn off OneAgent Metadata enrichment. If no "
+                             "OneAgent is running on the machine, this is "
+                             "ignored. Otherwise, OneAgent metadata will be "
+                             "added to each of the exported metric lines")
 
-requests_size = meter.create_valuerecorder(
-    name="request_size",
-    description="size of requests",
-    unit="1",
-    value_type=int,
-)
+    parser.set_defaults(metadata_enrichment=True)
+    return parser.parse_args()
 
 
-# Labels are used to identify key-values that are associated with a specific
-# metric that you want to record. These are useful for pre-aggregation and can
-# be used to store custom dimensions pertaining to a metric
-staging_labels = {"environment": "staging"}
-testing_labels = {"environment": "testing"}
+if __name__ == '__main__':
+    args = parse_arguments()
 
-# Update the metric instruments using the direct calling convention
-requests_counter.add(25, staging_labels)
-requests_size.record(100, staging_labels)
-time.sleep(10)
+    # set up opentelemetry for export:
+    metrics.set_meter_provider(MeterProvider())
+    meter = metrics.get_meter(__file__)
 
-requests_counter.add(50, staging_labels)
-requests_size.record(5000, staging_labels)
-time.sleep(5)
+    exporter = DynatraceMetricsExporter(args.endpoint, args.token,
+                                        export_oneagent_metadata=
+                                        args.metadata_enrichment)
 
-requests_counter.add(35, testing_labels)
-requests_size.record(2, testing_labels)
+    # this line registers the meter and exporter with the MeterProvider set
+    # above. All instruments created by the meter that is registered here will
+    # export to the Dynatrace metrics exporter. It is a good idea to keep a 
+    # reference to the meter (e. g. in a global variable) in order to create
+    # instruments anywhere in the code that all export to the same Dynatrace
+    # metrics exporter.
+    metrics.get_meter_provider().start_pipeline(meter, exporter, 5.)
```

Comment:
You could also make the export interval configurable. I'd suggest 10-60 secs in the arg parser and would make 10 secs the default.